src/domain/services/query.service.spec.ts

```diff
@@ -205,6 +205,81 @@ describe('count', () => {
       expect(response).toBe(3);
     });
   });
+
+  it('should throw an unprocessable exception if an $in contains an array', async () => {
+    await createTestingModule([DomainModule], async (app) => {
+      try {
+        await app
+          .get(QueryService)
+          .count({ origins_tags: { $in: ['a', ['b', 'c']] } });
+        fail('should not get here');
+      } catch (e) {
+        expect(e).toBeInstanceOf(UnprocessableEntityException);
+      }
+    });
+  });
+
+  it('should cope with an $in unknown value', async () => {
+    await createTestingModule([DomainModule], async (app) => {
+      const { originValue } = await createTestTags(app);
+      const queryService = app.get(QueryService);
+      const response = await queryService.count({
+        origins_tags: originValue,
+        nucleotides_tags: { $in: [null, []] },
+      });
+      expect(response).toBe(1);
+    });
+  });
+
+  it('should cope with an $in unknown value on a product field', async () => {
+    await createTestingModule([DomainModule], async (app) => {
+      const { originValue } = await createTestTags(app);
+      const queryService = app.get(QueryService);
+      const response = await queryService.count({
+        origins_tags: originValue,
+        creator: { $in: [null, []] },
+      });
+      expect(response).toBe(1);
+    });
+  });
+
+  it('should cope with $nin', async () => {
+    await createTestingModule([DomainModule], async (app) => {
+      const { originValue, aminoValue, aminoValue2 } = await createTestTags(
+        app,
+      );
+      const queryService = app.get(QueryService);
+      const response = await queryService.count({
+        origins_tags: originValue,
+        amino_acids_tags: { $nin: [aminoValue, aminoValue2] },
+      });
+      expect(response).toBe(0);
```

Comment:
Returning 0 result might be for other reasons than those expected.

Maybe add another test with `$nin: [aminoValue]` just to be sure the search work ?